File: ruby/lib/enolib/elements/missing/missing_value_element_base.rb
# frozen_string_literal: true

module Enolib
  class MissingValueElementBase < MissingElementBase
    def optional_value(_loader)
      nil
    end

    def optional_string_value
      nil
    end

    def required_string_value
      @parent._missing_error(self)
    end

    def required_value(_loader)
      @parent._missing_error(self)
    end
  end
end

File: python/enolib/elements/missing/missing_value_element_base.py
from .missing_element_base import MissingElementBase

class MissingValueElementBase(MissingElementBase):
    def optional_string_value(self):
        return None

    def optional_value(self, _loader):
        return None

    def required_string_value(self):
        self._parent._missing_error(self)

    def required_value(self, _loader):
        self._parent._missing_error(self)
